src/Sonarr.Http/Authentication/ApiKeyAuthenticationHandler.cs

```diff
@@ -61,11 +66,15 @@ protected override Task<AuthenticateResult> HandleAuthenticateAsync()
                 return Task.FromResult(AuthenticateResult.NoResult());
             }
 
-            if (_apiKey == providedApiKey)
+            // If the authentication method is set to None, this means the app has not been setup yet and allow all requests until setup is complete.
```

Comment:
Is this actually desired behavior, or should we make the setup mandatory?

Reply:
From the UI it will be mandatory, but we should have some sort of default user created with an API key that the user can change in that setup instead of ignoring the API key.

Reply:
Hi, @markus101 is this the direction you want to go towards?

Correct me if I'm wrong but isn't the authentication method only ever set to None before the app has been setup. So if this is because of security vulnerabilities, don't see how is any different from my method. 

In conclusion, i don't really see the point to doing this.

> default user created with an API key that the user can change in that setup instead of ignoring the API key

Could you please provide some reasons why your method is preferred compared to mine? Thanks!

Reply:
Ideally we'd never have Sonarr publicly exposed at all, this just makes that more obvious. `External` is going to have the same problem, there isn't an internal Sonarr configured, but that's a larger problem to solve.